libleptongui/src/x_menus.c

```diff
@@ -306,33 +146,18 @@ get_main_popup (GschemToplevel* w_current)
 
     /* No action --> add a separator */
     if (e.action == NULL) {
-#ifdef ENABLE_GTK3
-      menu_item = gtk_separator_menu_item_new ();
-#else
-      menu_item = gtk_menu_item_new();
-#endif
+      menu_item = make_separator_menu_item ();
       gtk_widget_show (menu_item);
       gtk_menu_shell_append (GTK_MENU_SHELL (menu), menu_item);
       continue;
     }
 
     /* Don't bother showing keybindings in the popup menu */
-    action = GSCHEM_ACTION (g_object_new (GSCHEM_TYPE_ACTION,
-                                          "name", e.action,
-                                          "label", gettext (e.name),
-                                          "tooltip", gettext (e.name),
-                                          NULL));
-    /* If there's a matching stock item, use it. Otherwise lookup the
-       name in the icon theme. */
-    if (e.stock_id != NULL && gtk_stock_lookup (e.stock_id, &stock_info)) {
-      gtk_action_set_stock_id (GTK_ACTION (action), e.stock_id);
-    } else {
-      gtk_action_set_icon_name (GTK_ACTION (action), e.stock_id);
-    }
-
-    /* Connect things up so that the actions get run */
-    g_signal_connect (G_OBJECT (action), "activate",
-                      G_CALLBACK (g_menu_execute), w_current);
+    action = make_menu_action (e.action,
```

Comment:
`unused variable 'stock_info' [-Wunused-variable]` in `get_main_popup()`

Reply:
Need to fix this, too.